kotsadm/pkg/handlers/identity.go

```diff
@@ -33,50 +35,67 @@ type IDPConfig struct {
 	OIDCConfig    *OIDCConfig `json:"oidcConfig"`
 	GEOAxISConfig *OIDCConfig `json:"geoAxisConfig"`
 }
-
 type OIDCConfig struct {
-	ConnectorID               string   `json:"connectorId"`
-	ConnectorName             string   `json:"connectorName"`
-	Issuer                    string   `json:"issuer"`
-	ClientID                  string   `json:"clientId"`
-	ClientSecret              string   `json:"clientSecret"`
-	GetUserInfo               *bool    `json:"getUserInfo"`
-	UserNameKey               string   `json:"userNameKey"`
-	InsecureSkipEmailVerified *bool    `json:"insecureSkipEmailVerified"`
-	InsecureEnableGroups      *bool    `json:"insecureEnableGroups"`
-	Scopes                    []string `json:"scopes"`
+	ConnectorID               string           `json:"connectorId"`
+	ConnectorName             string           `json:"connectorName"`
+	Issuer                    string           `json:"issuer"`
+	ClientID                  string           `json:"clientId"`
+	ClientSecret              string           `json:"clientSecret"`
+	GetUserInfo               *bool            `json:"getUserInfo"`
+	UserNameKey               string           `json:"userNameKey"`
+	UserIDKey                 string           `json:"userIDKey"`
+	PromptType                string           `json:"promptType"`
+	InsecureSkipEmailVerified *bool            `json:"insecureSkipEmailVerified"`
+	InsecureEnableGroups      *bool            `json:"insecureEnableGroups"`
+	Scopes                    []string         `json:"scopes"`
+	HostedDomains             []string         `json:"hostedDomains"`
+	ClaimMapping              OIDCClaimMapping `json:"claimMapping"`
+}
+type OIDCClaimMapping struct {
+	PreferredUsernameKey string `json:"preferredUsername"`
+	EmailKey             string `json:"email"`
+	GroupsKey            string `json:"groups"`
 }
 
 func ConfigureIdentityService(w http.ResponseWriter, r *http.Request) {
 	request := ConfigureIdentityServiceRequest{}
 	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
+		err = errors.Wrap(err, "failed to decode request body")
+		logger.Error(err)
+		JSON(w, http.StatusBadRequest, NewErrorResponse(err))
+		return
+	}
+
+	if err := validateConfigureIdentityRequest(request); err != nil {
+		err = errors.Wrap(err, "failed to validate request")
 		logger.Error(err)
-		response := ErrorResponse{Error: "failed to decode request body"}
-		JSON(w, http.StatusInternalServerError, response)
+		JSON(w, http.StatusBadRequest, NewErrorResponse(err))
 		return
 	}
 
 	namespace := os.Getenv("POD_NAMESPACE")
 
 	previousConfig, err := identity.GetConfig(r.Context(), namespace)
 	if err != nil {
-		logger.Error(errors.Wrap(err, "failed to get identity config"))
-		w.WriteHeader(http.StatusInternalServerError)
+		err = errors.Wrap(err, "failed to get identity config")
+		logger.Error(err)
+		JSON(w, http.StatusInternalServerError, NewErrorResponse(err))
```

Comment:
i dont think internal server errors should be json. it may reveal too much about what happened on the server

```go
err = errors.Wrap(err, "failed to get identity config")
logger.Error(err)
w.WriteHeader(http.StatusInternalServerError)
return
```